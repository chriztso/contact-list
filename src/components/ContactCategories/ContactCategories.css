.contact-categories{
    border: 1px solid rgb(233, 234, 240);
    display: flex; 
    padding: 5px 0px;     
    align-items: center;
    color: rgb(96, 102, 124);
    font-weight: 600;
}

.contact-categories__checkbox{
    margin-left: 12px;
    margin-right: 12px;
}

.contact-categories__contact-title{
    /* border: 1px solid white; */
    width: 20%; 
}

.contact-categories__value-title{
    /* border: 1px solid white; */
    width: 12%; 
    display: flex;
    align-items: center;
}

.contact-categories__location-title{
    /* border: 1px solid white; */
    width: 15%; 
}

.contact-categories__deals-title{
    /* border: 1px solid white; */
    width: 10%; 
}

.contact-categories__tags-title{
    /* border: 1px solid black; */
    width: 35%; 
}