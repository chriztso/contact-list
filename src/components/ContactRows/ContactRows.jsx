import Contact from '../Contact/Contact'
import { getStateAbbreviation } from '../../utils/format'   
import convertToUSD from '../../utils/convert'

const ContactRows = ( props ) => {
    const {
        contacts,
        dealsObject,
        tagsObject,
        contactTagsObject,
        geoIpsObject,
        geoAddressesObject
    } = props

    const getContactTags = (contactTags) => {
        let contactTagsArr = []
        //go through each contactTag ID in contact property array, 
        contactTags.forEach(contactTagID => {
            //access the object in contactTagObjects with contact tag ID
            const contactTagObject = contactTagsObject[contactTagID]
            //get tag ID from contactTag object 
            const tagId = contactTagObject["tag"]
            //access the object in tagObjects
            const tagObject = tagsObject[tagId]
            //get tag from tag object
            const tag = tagObject["tag"]
            //push to array
            contactTagsArr.push(tag)
        })
        console.log('CT ARR', contactTagsArr)
        //create string from contact tags array
        const contactTagsStr = contactTagsArr.join(', ')
        console.log("CT STR", contactTagsStr)
        return contactTagsStr
    }

    const getDealsTotalValue = (deals) => {
        let dealsTotalValueUSD = 0
        //go through each deal ID in contact property array
        deals.forEach(dealID => {
            //access the deal object in dealObjects with each deal ID
            const dealObj = dealsObject[dealID]
            //get value and currency of deal object and add to dealsTotalValue
            const { value, currency } = dealObj
            //convert to USD if currency is not USD
            if(currency !== "usd"){
              const valueInUSD = convertToUSD(currency, Number(value))
              dealsTotalValueUSD += valueInUSD
            } else {
              //add without conversion if in USD
              dealsTotalValueUSD += Number(value)
            }
        })
        console.log("TOTAL VAL", dealsTotalValueUSD)
        //format number with commas
        dealsTotalValueUSD = dealsTotalValueUSD.toLocaleString("en-US")
        return dealsTotalValueUSD
    }

    const getLocation = (geoIps) => {
        console.log("GE", geoIps)
        let location = ''
        //go through each geoIps ID 
        geoIps.forEach(geoIpsID => {
            //access geoIps object in geoIpsObjects with each geoIps ID
            const geoIpsObj = geoIpsObject[geoIpsID]
            //accesss the geoAddrID property in each object
            const geoAddrID = geoIpsObj["geoaddrid"]
            //access geoAddr object in geoAddressObjects with geoAddr ID
            const geoAddressObject = geoAddressesObject[geoAddrID]
            //get city, state, country
            const country = geoAddressObject["country2"]
            const city = geoAddressObject["city"]
            const stateAbbreviation = getStateAbbreviation(geoAddressObject["state"])
            location = `${city}, ${stateAbbreviation}, ${country}`
            console.log("LOCATION", location)
        })
        return location
    }

    return (
        <div className="contact-rows">
            {contacts.map((contact) => {
                const { firstName, lastName, geoIps, contactTags, deals } = contact
                console.log('___________________________________')
                const name = `${firstName} ${lastName}`
                const initials = `${firstName[0]}${lastName[0]}`
                console.log('NAME', name)
                const contactTagsStr = getContactTags(contactTags)
                const numberOfDeals = deals.length
                const dealsTotalValue = getDealsTotalValue(deals)

                let geoAddressStr = ""
                console.log("GEO", geoIps.length)
                if(geoIps.length === 0){
                    geoAddressStr = 'N/A'
                } else {
                  geoAddressStr = getLocation(geoIps)
                  console.log('GA ARR', geoAddressStr)
                  console.log('___________________________________')
                }

                return (
                  <Contact 
                      name={name}
                      initials={initials}
                      contactTagsStr={contactTagsStr}
                      numberOfDeals={numberOfDeals}
                      dealsTotalValue={dealsTotalValue}
                      geoAddressStr={geoAddressStr}
                  />
                )
            })}
        </div>
    )
}

export default ContactRows