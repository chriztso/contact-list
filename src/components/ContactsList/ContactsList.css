.contacts-list{
    border: 1px solid rgb(242, 243, 247); 
}