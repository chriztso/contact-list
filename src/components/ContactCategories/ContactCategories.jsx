import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
import MoreHorizIcon from '@material-ui/icons/MoreHoriz';
import './ContactCategories.css'

const ContactCategories = () => {
    return (
        <div className="contact-categories">
            <input type="checkbox" className="contact-categories__checkbox"></input>
            <span className="contact-categories__contact-title">Contact</span>
            <span className="contact-categories__value-title">
                Total Value
                <ArrowDropDownIcon />
            </span>
            <span className="contact-categories__location-title">Location</span>
            <span className="contact-categories__deals-title">Deals</span>
            <span className="contact-categories__tags-title">Tags</span>
            <MoreHorizIcon />
        </div>
    )
}

export default ContactCategories