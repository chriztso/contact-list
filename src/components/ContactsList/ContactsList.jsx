import './ContactsList.css'
import ContactCategories from '../ContactCategories/ContactCategories'
import ContactRows from '../ContactRows/ContactRows'

const ContactsList = (props) => {
    const {
        contacts,
        dealObjects,
        tagObjects,
        contactTagObjects,
        geoIpsObjects,
        geoAddressObjects
    } = props

    return (
        <div className="contacts-list">
            <ContactCategories />
            <ContactRows
                contacts={contacts}
                dealObjects={dealObjects}
                tagObjects={tagObjects}
                contactTagObjects={contactTagObjects}
                geoIpsObjects={geoIpsObjects}
                geoAddressObjects={geoAddressObjects}
            />
        </div>
    )
}

export default ContactsList