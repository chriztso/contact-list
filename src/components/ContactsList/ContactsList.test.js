import { shallow } from 'enzyme';
import ContactsList from '../ContactsList/ContactsList'
import ContactCategories from '../ContactCategories/ContactCategories'
import ContactRows from '../ContactRows/ContactRows'

describe('ContactsList', () => {
    it('renders ContactsList', () => {
        const wrapper = shallow(<ContactsList />);
        expect(wrapper).toMatchSnapshot();
    })

    it('renders ContactCategories', () => {
       const wrapper = shallow(<ContactsList />);
        expect(wrapper.find(ContactCategories).length).toEqual(1);
    })

    it('renders ContactRows', () => {
        const wrapper = shallow(<ContactsList />);
        expect(wrapper.find(ContactRows).length).toEqual(1);
    })
})