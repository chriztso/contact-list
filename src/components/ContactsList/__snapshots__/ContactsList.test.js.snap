// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsList renders ContactsList 1`] = `
<div
  className="contacts-list"
>
  <ContactCategories />
  <ContactRows />
</div>
`;
